bugsnag-android-core/src/test/java/com/bugsnag/android/BreadcrumbStateTest.kt

```diff
@@ -121,8 +121,50 @@ class BreadcrumbStateTest {
      */
     @Test
     fun testOnBreadcrumbCallback() {
-        breadcrumbState.callbackState.addOnBreadcrumb(OnBreadcrumbCallback { false })
-        breadcrumbState.add(Breadcrumb("Whoops"))
+        val breadcrumb = Breadcrumb("Whoops")
+        breadcrumbState.callbackState.addOnBreadcrumb(OnBreadcrumbCallback {
+            true
+        })
+        breadcrumbState.add(breadcrumb)
+        assertEquals(1, breadcrumbState.store.size)
+        assertEquals(breadcrumb, breadcrumbState.store.peek())
+    }
+
+    /**
+     * Verifies that returning false in one callback will halt subsequent callbacks and not apply breadcrumb
+     */
+     @Test
+     fun testOnBreadcrumbCallbackFalse() {
+        val breadcrumb = Breadcrumb("Whoops")
+        breadcrumbState.callbackState.addOnBreadcrumb(OnBreadcrumbCallback { givenBreadcrumb ->
```

Comment:
Assuming I understand the requirements correctly, a slightly better test would be to have an extra callback before this that returns true, verifying that a single entry makes it into the store.